solutions/disconnected_installs/docker-registry-sync.py

```diff
@@ -0,0 +1,95 @@
+parser.add_argument('--file', action='store', dest='json_file', help='A JSON formatted file with the following format:'
+                                                                     '{"<tag_type>": {"<namespace>": ["image1", image2"'
+                                                                     'image3]}}')
+parser.add_argument('--output', action='store_true', dest='output', help='If this flag is present, commands will be'
+                                                                         'dumped to stdout instead of run')
+
+options = parser.parse_args()
+
+release_version = '3.4'
+
+retrieve_v_tags_from_redhat_list = []
+retrieve_non_v_tags_from_redhat_list = []
+
+latest_tag_list = []
+
+def generate_url_list(dictionary_key, list_to_populate):
+    for namespace in config_file_dict[dictionary_key]:
+        for image in config_file_dict[dictionary_key][namespace]:
+            docker_json_link = "https://registry.access.redhat.com/v2/%s/%s/tags/list" % (namespace, image)
+            list_to_populate.append(docker_json_link)
+
+def get_latest_tag_from_api(url_list, tag_list, version_type = None):
+    for url in url_list:
+        redhat_registry = urllib.urlopen(url)
+        # The object is returned as a string so it needs to be converted to a json object
+        image_tag_dictionary = json.loads(redhat_registry.read())
+        # Get the latest version for a given release
+        latest_tag = ''
+        image_name = image_tag_dictionary['name']
+        for tag in image_tag_dictionary['tags']:
+            # check to see if there is a 'v' in the version tag:
+            if tag.startswith('v'):
+                # This tracks the position of the splice. It assumes that you are trying to get the latest
+                # release based on a two digit release (i.e. 3.4 or 3.7)
+                splice_position = 4
+            else:
+                splice_position = 3
+            if release_version in tag[:splice_position]:
+                # There may be a better way of getting the highest tag for a release
+                # but the list may potentially have a higher release version than what you are looking for
+                if parse_version(tag) > parse_version(latest_tag):
+                    if version_type is not None:
+                        if "v" in tag:
+                            pass
+                        else:
+                            latest_tag = tag
+                    else:
+                        latest_tag = tag
+        # We want to remove everything after the hyphen because we don't care about release versions
+        latest_tag_minus_hyphon = latest_tag.split('-')[0]
+        tag_list.append("%s:%s" % (image_name, latest_tag_minus_hyphon))
+
+
+config_file = options.json_file
+with open(config_file) as json_data:
+    config_file_dict = json.load(json_data)
+
+generate_url_list('image_with_v_in_tag', retrieve_v_tags_from_redhat_list)
```

Comment:
RE: categories, I can change this fairly easily 

RE: s2i images, we can do a category just for S2I because they dont really relate tot he OpenShift release version as I understand. I chose json files so that it was easy to add new categories